src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side must be greater than zero');
+    }
+
+    if ((a + b <= c) || (a + c <= b) || (b + c <= a)) {
+      throw new Error('The hypotenuse is shorter than the sum of the legs');
+    }
+  }
+
+  getArea(): number {
+    const halfOfPerimeter: number = (this.a + this.b + this.c) / 2;
 
+    const triangleArea = Math.sqrt(
+      halfOfPerimeter
+      * (halfOfPerimeter - this.a)
+      * (halfOfPerimeter - this.b)
+      * (halfOfPerimeter - this.c),
+    );
+
+    return Math.floor(triangleArea * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius should be greater then zero');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * (this.radius ** 2) * 100) / 100;
+  }
 }
 
 export class Rectangle {
+  shape: Shape = 'rectangle';
 
-}
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (height <= 0 || width <= 0) {
+      throw new Error('Sides must be greater than zero');
+    }
+  }
 
-export function getInfo(figure) {
+  getArea():number {
+    return this.width * this.height;
```

Comment:
I was advised to round these values, so you can also do that for improvement (optional)